### PR TITLE
Update to use getDerivedStateFromProps

### PR DESCRIPTION
**Pull request by chrismllr**
Also stores the linkifyit instance within state, instead of on the Class.

Resolves issue https://github.com/obipawan/react-native-hyperlink/issues/27

### CODE REVIEW

**Comment by Linoa65** [comment]
Some news about this PR ? React warn us about deprecated usage, this PR seems to be ok, why this is unmerged ?

**Comment by karltaylor** [comment]
Also getting this warning. Might have to patchPackage this until this is merged but would be great to get it in @obipawan

**Comment by bitcrumb** [comment]
@obipawan Can this PR be taken up & merged? Thanks! Seems like a reasonably sized PR ;-)

**Comment by stianjensen** [comment]
The PR has a conflict that needs to be solved, but it would be great to get this in now that the latest version of react native gives a warning for using componentWillReceiveProps.

**Comment by chrismllr** [comment]
Resolved the conflicts, should be good to go 👍

**Comment by ShotSkydiver** [comment]
There actually were still a few issues with this PR's changes; running `npm run build` fails with a babel linting error, I've created a PR in chrismllr's forked repo, so it can be merged into his fork and then this PR can automatically reflect my fixes!

File: src/Hyperlink.js [hunk 1]
```diff
@@ -52,10 +59,10 @@ class Hyperlink extends Component {
     return typeof component.props.children !== 'string'
   }
 
-  linkify(component){
+  linkify = (component) => {
```
Comment (ShotSkydiver) [inline src/Hyperlink.js]: The babel build script failed due to this change; transforming this new class property syntax requires the babel plugin [proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), which requires Babel 7

**Comment by chrismllr** [comment]
@ShotSkydiver Yeah, that makes sense. Just merged your PR into my fork

**Comment by obipawan** [comment]
Hi @chrismllr @ShotSkydiver 
Thanks for the PR ❤️ 🎊 
My sincere apologies to all those who've been waiting for this. I'm getting this in with the next release

**Comment by obipawan** [comment]
Available with [v0.0.17](https://github.com/obipawan/react-native-hyperlink/releases/tag/v0.0.17
) 👍

File: src/Hyperlink.js [hunk 2]
```diff
@@ -16,15 +16,22 @@ const textPropTypes = Text.propTypes || {}
 const { OS } = Platform
 
 class Hyperlink extends Component {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.linkify !== prevState.linkifyIt) {
```
[the remark this answers is not in the document]
Comment (stianjensen) [reply src/Hyperlink.js]: Testing this locally now, it seems this change has a problem if you do not pass in the `linkify` prop from outside. That will trigger a getDerivedStateFromProps where nextProps.linkify is undefined, and prevState.linkifyIt is the default value from the constructor (`require('linkify-it')()`), and thereby unset the state variable.

**Pull request merged**